pkgs/tools/virtualization/ovftool/default.nix

```diff
@@ -1,149 +1,234 @@
-      --set-default LC_CTYPE C.UTF-8 \
-      --prefix LD_LIBRARY_PATH : "$out/lib"
+    makeWrapper "$out/libexec/ovftool" "$out/bin/ovftool" \
+  '' + lib.optionalString stdenv.isLinux ''
+    --prefix LD_LIBRARY_PATH:"$out/lib"
+  '' + lib.optionalString stdenv.isDarwin ''
+    --prefix DYLD_LIBRARY_PATH:"$out/lib"
+  '' + ''
     runHook postInstall
   '';
 
-  preFixup = ''
+  preFixup = lib.optionalString stdenv.isLinux ''
     addAutoPatchelfSearchPath "$out/lib"
+  '' + lib.optionalString stdenv.isDarwin ''
+    change_args=()
+
+    # Change relative @loader_path dylibs to absolute paths.
+    for lib in $out/lib/*.dylib; do
+      libname=$(basename $lib)
+      change_args+=(-change "@loader_path/lib/$libname" "$out/lib/$libname")
+    done
+
+    # Patches for ovftool binary
+    change_args+=(-change /usr/lib/libSystem.B.dylib ${darwin.Libsystem}/lib/libSystem.B.dylib)
+    change_args+=(-change /usr/lib/libc++.1.dylib ${stdenv.cc.libcxx.cxxabi}/lib/libc++.1.dylib)
+    change_args+=(-change /usr/lib/libiconv.2.dylib ${libiconv}/lib/libiconv.2.dylib)
+    change_args+=(-change /usr/lib/libxml2.2.dylib ${libxml2}/lib/libxml2.2.dylib)
+    change_args+=(-change /usr/lib/libz.1.dylib ${zlib}/lib/libz.1.dylib)
+
+    change_args+=(-change @loader_path/lib/libcares.2.dylib ${c-ares}/lib/libcares.2.dylib)
+    change_args+=(-change @loader_path/lib/libexpat.dylib ${expat}/lib/libexpat.dylib)
+    change_args+=(-change @loader_path/lib/libicudata.60.2.dylib ${icu60}/lib/libicudata.60.2.dylib)
+    change_args+=(-change @loader_path/lib/libicuuc.60.2.dylib ${icu60}/lib/libicuuc.60.2.dylib)
+    change_args+=(-change @loader_path/lib/libxerces-c-3.2.dylib ${xercesc}/lib/libxerces-c-3.2.dylib)
+    # Patch binary
+    install_name_tool "''${change_args[@]}" "$out/libexec/ovftool"
+
+    # Additional patches for ovftool dylibs
+    change_args+=(-change /usr/lib/libresolv.9.dylib ${darwin.Libsystem}/lib/libresolv.9.dylib)
+    change_args+=(-change @loader_path/libcares.2.dylib ${c-ares}/lib/libcares.2.dylib)
+    change_args+=(-change @loader_path/libexpat.dylib ${expat}/lib/libexpat.dylib)
+    change_args+=(-change @loader_path/libicudata.60.2.dylib ${icu60}/lib/libicudata.60.2.dylib)
+    change_args+=(-change @loader_path/libicuuc.60.2.dylib ${icu60}/lib/libicuuc.60.2.dylib)
+    change_args+=(-change @loader_path/libxerces-c-3.2.dylib ${xercesc}/lib/libxerces-c-3.2.dylib)
+
+    # Add new abolute paths for other libs to all libs
+    for lib in $out/lib/*.dylib; do
+      libname=$(basename $lib)
+      change_args+=(-change "@loader_path/$libname" "$out/lib/$libname")
+    done
+
+    # Patch all libs
+    for lib in $out/lib/*.dylib; do
+      libname=$(basename $lib)
+      install_name_tool -id "$libname" "$lib"
+      install_name_tool "''${change_args[@]}" "$lib"
+    done
   '';
 
-  doInstallCheck = true;
+  # Tests fail on darwin as ovftool tries many things that breaks the sandbox.
```

Comment:
Is there any way we could do a more limited installCheck for Darwin?

Reply:
I was trying, but ovftool is using some lib that seems to be doing more than one would expect. I'll run the InstallCheck again and see if I can get a list of paths.

Reply:
These are the logs from Sandboxd durring the InstallCheck. Notably, it's trying to read `/private/etc/passwd`

```
error	12:40:21.842670-0800	kernel	Sandbox: bash(48538) deny(1) sysctl-write kern.system_version_compat
error	12:40:21.844640-0800	kernel	Sandbox: bash(48538) deny(1) file-read-data /private/etc/passwd
error	12:40:21.844721-0800	kernel	Sandbox: bash(48538) deny(1) file-read-metadata /private/etc/passwd
error	12:40:21.845894-0800	kernel	Sandbox: bash(48538) deny(1) mach-lookup com.apple.system.notification_center
error	12:40:21.857345-0800	kernel	Sandbox: bash(48538) deny(1) mach-lookup com.apple.diagnosticd
error	12:40:21.868379-0800	kernel	Sandbox: bash(48538) deny(1) file-read-data /private/etc/passwd
error	12:40:21.868424-0800	kernel	Sandbox: bash(48538) deny(1) file-read-metadata /private/etc/passwd
error	12:40:21.869431-0800	kernel	Sandbox: bash(48538) deny(1) mach-lookup com.apple.diagnosticd
error	12:40:25.883889-0800	kernel	Sandbox: ovftool(48538) deny(1) file-read-data /dev/autofs_nowait
error	12:40:25.904944-0800	kernel	Sandbox: ovftool(48538) deny(1) file-read-metadata /private/var/db/timezone/zoneinfo
error	12:40:25.905636-0800	kernel	4 duplicate reports for Sandbox: ovftool(48538) deny(1) file-read-metadata /private/var/db/timezone/zoneinfo
error	12:40:25.905646-0800	kernel	Sandbox: ovftool(48538) deny(1) file-read-data /private/etc/passwd
error	12:40:25.905668-0800	kernel	Sandbox: ovftool(48538) deny(1) file-read-metadata /private/etc/passwd
error	12:40:25.905790-0800	kernel	Sandbox: ovftool(48538) deny(1) mach-lookup com.apple.diagnosticd
error	12:40:25.906562-0800	kernel	Sandbox: ovftool(48538) deny(1) file-read-data /private/etc/passwd
error	12:40:25.906586-0800	kernel	Sandbox: ovftool(48538) deny(1) file-read-metadata /private/etc/passwd
error	12:40:25.906658-0800	kernel	Sandbox: ovftool(48538) deny(1) mach-lookup com.apple.diagnosticd
```

Reply:
ouch, that's annoying. Maybe reach out on the Discourse, there are people much more adept at Darwin packaging than I.

Reply:
Got the tests working. Had to expose a few more paths in the sandbox and make a dummy /etc/passwd.